doc/specs/#2046 - Unified keybindings and commands, and synthesized action names.md

```diff
@@ -0,0 +1,589 @@
+  pressing that keybinding should open the Command Palette directly to that
+  level of nesting of commands.
+* **"Iterable" commands** (with an `iterateOn` property): These are commands
+  that are expanded into one command per profile. These cannot really be bound
+  as keybindings - which action should be bound to the key? They can't all be
+  bound to the same key. If a KeyBinding/Command json blob has a valid
+  `iterateOn` property, then we'll ignore it as a keybinding. This includes any
+  commands that are nested as children of this command - we won't be able to
+  know which of the expanded children will be the one to bind the keys to.
+
+<sup>[1]</sup>: This requirement will be relaxed given **Proposal 2**, below,
+but ignored for the remainder of this section, for illustrative purposes.
+
+#### Example
+
+Consider the following settings:
+
+```json
+"commands": [
+  { "name": "Duplicate Tab", "command": "duplicateTab", "keys": "ctrl+alt+a" },
+  { "command": "nextTab", "keys": "ctrl+alt+b" },
+  {
+    "icon": "...",
+    "name": { "key": "NewTabWithProfileRootCommandName" },
+    "commands": [
+      {
+        "iterateOn": "profiles",
+        "icon": "${profile.icon}",
+        "name": { "key": "NewTabWithProfileCommandName" },
+        "command": { "action": "newTab", "profile": "${profile.name}" }
+      }
+    ]
+  },
+  {
+    "icon": "...",
+    "name": "Connect to ssh...",
+    "commands": [
+      {
+        "keys": "ctrl+alt+c",
+        "icon": "...",
+        "name": "first.com",
+        "command": { "action": "newTab", "commandline": "ssh me@first.com" }
+      },
+      {
+        "keys": "ctrl+alt+d",
+        "icon": "...",
+        "name": "second.com",
+        "command": { "action": "newTab", "commandline": "ssh me@second.com" }
+      }
+    ]
+  }
+  {
+    "keys": "ctrl+alt+e",
+    "icon": "...",
+    "name": { "key": "SplitPaneWithProfileRootCommandName" },
+    "commands": [
+      {
+        "iterateOn": "profiles",
+        "icon": "${profile.icon}",
+        "name": { "key": "SplitPaneWithProfileCommandName" },
+        "commands": [
+          {
+            "keys": "ctrl+alt+f",
+            "icon": "...",
+            "name": { "key": "SplitPaneName" },
+            "command": { "action": "splitPane", "profile": "${profile.name}", "split": "automatic" }
+          },
+          {
+            "icon": "...",
+            "name": { "key": "SplitPaneVerticalName" },
+            "command": { "action": "splitPane", "profile": "${profile.name}", "split": "vertical" }
+          },
+          {
+            "icon": "...",
+            "name": { "key": "SplitPaneHorizontalName" },
+            "command": { "action": "splitPane", "profile": "${profile.name}", "split": "horizontal" }
+          }
+        ]
+      }
+    ]
+  }
+]
+```
+
+> 👉 NOTE: The above could have just as easily been placed in `keybindings`, or
+> split across both `keybindings` and `commands`.

```

Comment:
Yea, that's where the complication comes in. I think that `commands` is the more future-proof name, but at this point, we can't really deprecate the `keybindings` name anymore. So we could _start_ using `commands` in the future for everything, and change `defaults.json` to have a `commands` instead of a `keybindings`, etc, but we'll _always_ need to keep `keybindings` around.

I'd prefer to move to just `commands`, but I figured I'd make sure to leave notes that `keybindings` would still work as a property key.

Reply:
Just throwing this out there as an idea, when we start adding bindings for mouse/touch/stylus, would this still be fine being named `"commands"`? Or should we consider something like `"bindings"`?

I'm fine with either, just wanted your thoughts.

Reply:
Whoops. Just noticed that Dustin made the same point down in his review.